### PR TITLE
Adding the object orientation based manipulation reward and observation

### PR DESCRIPTION
**Pull request by jtigue-bdai**
# Description

This PR adds a new reward and observation for rigid objects during manipulation tasks.

Fixes # (issue)

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/lift/mdp/observations.py [hunk 1]
```diff
@@ -8,9 +8,9 @@
 import torch
 from typing import TYPE_CHECKING
 
-from isaaclab.assets import RigidObject
+from isaaclab.assets import Articulation, RigidObject
 from isaaclab.managers import SceneEntityCfg
-from isaaclab.utils.math import subtract_frame_transforms
+from isaaclab.utils.math import matrix_from_quat, quat_inv, quat_mul, subtract_frame_transforms
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/lift/mdp/observations.py:13]: Maybe this is simpler?

```python
import isaaclab.utils.math as math_utils
```

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/lift/mdp/observations.py [hunk 2]
```diff
@@ -22,8 +22,36 @@ def object_position_in_robot_root_frame(
...
+        rotation matrices representing the object's orientation in the robot's root frame
+        for each environment.
+    """
+    robot: Articulation = env.scene[robot_cfg.name]
+    object: RigidObject = env.scene[object_cfg.name]
+    object_quat_w = object.data.root_quat_w[:, 0:4]
+    robot_quat_w = robot.data.root_quat_w[:, 0:4]
+    return (
+        matrix_from_quat(quat_mul(quat_inv(robot_quat_w), object_quat_w))[:, 0:3, 0:3]
+        .permute(0, 2, 1)
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/lift/mdp/observations.py:55]: why do you need permute operation? And can you use view instead of rehape?

**Review by Mayankm96 — CHANGES_REQUESTED** [review]
Not sure how useful it is without an example using these observations and rewards.
